### PR TITLE
Fully qualify C++ types without explicit namespace

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
Depends on #858

### CODE REVIEW

**Comment by ahayzen-kdab** [comment]
@LeonMatthesKDAB leave as draft if a pull request depends / includes another, otherwise folks may review the wrong one first :-)

**Comment by LeonMatthesKDAB** [comment]
Hm, I kind of hate how inconsistent this has become...
Whenever we hard-code C++ into our generator we still tend to use fully-qualified paths (which I think is still the best approach).
But for every generated path, we no use paths that are not quite fully qualified...

That's really unfortunate :(
But if it works I guess it must do...

File: crates/cxx-qt-gen/src/naming/type_names.rs [hunk 1]
```diff
@@ -178,6 +178,15 @@ impl TypeNames {
     }
 
     /// For a given rust ident return the CXX name with its namespace
+    ///
+    /// Ideally we'd want this type name to always be **fully** qualified, staring with `::`.
+    /// Unfortunately, this isn't always possible, as the Qt5 meta object system doesn't register
+    /// types with the fully qualified path :(
+    /// E.g. it will recognize `QString`, but not `::QString` from QML.
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/naming/type_names.rs:185]: hmm what we could possibly do, any type that is registered in a C++ block we could do the required Qt registration in our C++ generation :thinking: 

Eg if there is a `type T = ...` then we could generate the missing type declaration with the fully qualified version. Guess the only fun would be if the type is not in fact supposed to be registered?

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/naming/type_names.rs:185]: Interesting idea...
At least we could do that for all QObject types that we found in `extern "C++Qt"` and `extern "RustQt"`.
Though as you said, unsure if there's ever a point where it **shouldn't** be registered :thinking: 

Maybe let's add this as a note into #619 and try this in another PR.

Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/naming/type_names.rs:185]: Commented :-)

**Review by ahayzen-kdab — APPROVED** [review]
Lets go with this to move forward and hope to tidy up the Qt 5 mess later.

**Pull request merged**